apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/Trace/ManagedTracerFactory.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright 2017 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Api.Gax;
+
+using TraceProto = Google.Cloud.Trace.V1.Trace;
+
+namespace Google.Cloud.Diagnostics.Common
+{
+    /// <summary>
+    /// A factory <see cref="ManagedTracerFactory"/> that will generate <see cref="IManagedTracer"/>s 
+    /// from <see cref="TraceHeaderContext"/>s.  If the <see cref="TraceHeaderContext"/> does not provide
+    /// the needed context to determine the proper <see cref="IManagedTracer"/> then the given
+    /// <see cref="ITraceOptionsFactory"/> will decide.
+    /// </summary>
+    internal class ManagedTracerFactory : IManagedTracerFactory
+    {
+        private readonly string _projectId;
+        private readonly IConsumer<TraceProto> _consumer;
+        private readonly ITraceOptionsFactory _optionsFactory;
+        private readonly TraceIdFactory _traceIdFactory;
+
+        /// <param name="projectId">The Google Cloud Platform project ID. Cannot be null.</param>
+        /// <param name="consumer">A trace consumer for the tracer. Cannot be null.</param>
+        /// <param name="optionsFactory">An options factory to fall back to if the 
+        ///     <see cref="TraceHeaderContext"/> does not provide enough context. Cannot be null.</param>
+        /// <param name="traceIdFactory">A trace Id factory. Cannot be null.</param>
```

Comment:
So I thought this kind of thing (having \<param\> tags without a \<summary\> tag) would cause a warning, so I tried this and it did not, but then I realized that the "XML documentation file" build option was not enabled for this project. Enabling that showed a bunch of warnings, but still this wasn't one of them.

That said, you should probably always have a \<summary\> tag if you're going to have the \<param\> tags.

Additionally, in a separate PR, you should enable the "XML documentation file" option for all non- test/snippet projects and then clean up all warnings produced by the build. You can either do this in the project properties > Build tab in VS, or by adding the `"xmlDoc": true` entry to the "buildOptions" in the project.json files.